### PR TITLE
Decoding of secrets rendered in the encoded user_data for EC2. Closes #261

### LINKED ISSUE #261: Decoding of secrets rendered in the encoded user_data for EC2.
**Is your feature request related to a problem? Please describe.**
Find secrets in EC2 User Data is one of the ISO-27001 requirement.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
Autoscaling - https://docs.aws.amazon.com/autoscaling/ec2/userguide/create-launch-config.html
EC2 - https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-add-user-data.html

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/aws_ec2_launch_configuration []

PRETEST: tests/aws_ec2_launch_configuration

TEST: tests/aws_ec2_launch_configuration
Running terraform
aws_ebs_volume.my_volume: Creating...
aws_ebs_volume.my_volume: Still creating... [10s elapsed]
aws_ebs_volume.my_volume: Creation complete after 14s [id=vol-0a2167ddb36ce908e]
aws_ebs_snapshot.my_snapshot: Creating...
aws_ebs_snapshot.my_snapshot: Creation complete after 3s [id=snap-09a75f883a51923ed]
aws_ami.named_test_resource: Creating...
aws_ami.named_test_resource: Still creating... [10s elapsed]
aws_ami.named_test_resource: Creation complete after 11s [id=ami-0ecf7b5ba1697e802]
aws_launch_configuration.named_test_resource: Creating...
aws_launch_configuration.named_test_resource: Creation complete after 3s [id=turbottest74833]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 4 added, 0 changed, 0 destroyed.

Outputs:

account_id = "013122550996"
aws_partition = "aws"
image_id = "ami-0ecf7b5ba1697e802"
resource_aka = "arn:aws:autoscaling:us-east-1:013122550996:launchConfiguration:869773ad-5aa4-4d20-bd2c-47927146b6e9:launchConfigurationName/turbottest74833"
resource_id = "turbottest74833"
resource_name = "turbottest74833"

Running SQL query: test-get-query.sql
[
  {
    "ebs_optimized": false,
    "image_id": "ami-0ecf7b5ba1697e802",
    "instance_monitoring_enabled": true,
    "instance_type": "t2.nano",
    "launch_configuration_arn": "arn:aws:autoscaling:us-east-1:013122550996:launchConfiguration:869773ad-5aa4-4d20-bd2c-47927146b6e9:launchConfigurationName/turbottest74833",
    "name": "turbottest74833"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "akas": [
      "arn:aws:autoscaling:us-east-1:013122550996:launchConfiguration:869773ad-5aa4-4d20-bd2c-47927146b6e9:launchConfigurationName/turbottest74833"
    ],
    "name": "turbottest74833",
    "title": "turbottest74833"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "launch_configuration_arn": "arn:aws:autoscaling:us-east-1:013122550996:launchConfiguration:869773ad-5aa4-4d20-bd2c-47927146b6e9:launchConfigurationName/turbottest74833",
    "name": "turbottest74833"
  }
]
✔ PASSED

POSTTEST: tests/aws_ec2_launch_configuration

TEARDOWN: tests/aws_ec2_launch_configuration

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select name,user_data from aws_ec2_launch_configuration where user_data like any (array['%pass%', '%secret%','%token%','%key%']) or user_data ~ '(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]'
+----------+---------------------------------------------------------------+
| name     | user_data                                                     |
+----------+---------------------------------------------------------------+
| testCopy | #!/bin/bash                                                   |
|          | yum update -y                                                 |
|          | amazon-linux-extras install -y lamp-mariadb10.2-php7.2 php7.2 |
|          | yum install -y httpd mariadb-server                           |
|          | systemctl start httpd                                         |
|          | systemctl enable httpd                                        |
|          | usermod -a -G apache ec2-user                                 |
|          | chown -R ec2-user:apache /var/www                             |
|          | chmod 2775 /var/www                                           |
|          | find /var/www -type d -exec chmod 2775 {} \;                  |
|          | find /var/www -type f -exec chmod 0664 {} \;                  |
|          | echo "<?php phpinfo(); ?>" > /var/www/html/phpinfo.php        |
|          | $key Test@123                                                 |
+----------+---------------------------------------------------------------+
> select instance_id,user_data from aws_ec2_instance where user_data like any (array['%pass%', '%secret%','%token%','%key%']) or user_data ~ '(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]'
+---------------------+---------------------------------------------------------------+
| instance_id         | user_data                                                     |
+---------------------+---------------------------------------------------------------+
| i-0017619f4ea5acf9c | #!/bin/bash                                                   |
|                     | yum update -y                                                 |
|                     | amazon-linux-extras install -y lamp-mariadb10.2-php7.2 php7.2 |
|                     | yum install -y httpd mariadb-server                           |
|                     | systemctl start httpd                                         |
|                     | systemctl enable httpd                                        |
|                     | usermod -a -G apache ec2-user                                 |
|                     | chown -R ec2-user:apache /var/www                             |
|                     | chmod 2775 /var/www                                           |
|                     | find /var/www -type d -exec chmod 2775 {} \;                  |
|                     | find /var/www -type f -exec chmod 0664 {} \;                  |
|                     | echo "<?php phpinfo(); ?>" > /var/www/html/phpinfo.php        |
|                     | echo "password - Test@123"                                    |
+---------------------+---------------------------------------------------------------+

```
</details>

### CODE REVIEW

File: aws/table_aws_ec2_launch_configuration.go [hunk 1]
```diff
@@ -146,6 +147,8 @@ func tableAwsEc2LaunchConfiguration(_ context.Context) *plugin.Table {
 				Description: "A list that contains the security groups to assign to the instances in the Auto Scaling group.",
 				Type:        proto.ColumnType_JSON,
 			},
+
+			/// Steampipe standard columns
```
Comment (Subhajit97) [inline aws/table_aws_ec2_launch_configuration.go:151]:
```suggestion
			// Steampipe standard columns
```

File: docs/tables/aws_ec2_instance.md [hunk 2]
```diff
@@ -31,7 +30,6 @@ where
   monitoring_state = 'disabled';
 ```
 
-
```
Comment (Subhajit97) [inline docs/tables/aws_ec2_instance.md:34]: Did we need to remove these lines? What do other docs examples have?

Comment (bigdatasourav) [reply docs/tables/aws_ec2_instance.md:34]: we always provide line space after query and comment , for other doc example also have same .

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**